### PR TITLE
adding env var for --terragrunt-include-external-dependencies

### PR DESCRIPTION
**Pull request by elebertus**
Hey there!

We've been running into the same issue as #1418 and [gruntwork-io/terratest/issues/699](https://github.com/gruntwork-io/terratest/issues/699). This small patch provides the ability to use an environment var with the expected behavior.

I looked through the tests around this code and didn't see where the env var bits for cli flags were tested. I'm happy to add coverage for this or discuss any required changes.

Let me know what thoughts are around the change.

Edit: forgot to include test run output, gist link [here](https://gist.github.com/elebertus/449c9b72e9a9310e920518996140c051) - not sure if the full integration tests need to pass as well, in which case I'll need to read through what is required in AWS and plan accordingly.

### CODE REVIEW

File: cli/args.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func parseTerragruntOptionsFromArgs(terragruntVersion string, args []string, wri
 
 	ignoreExternalDependencies := parseBooleanArg(args, OPT_TERRAGRUNT_IGNORE_EXTERNAL_DEPENDENCIES, false)
 
-	includeExternalDependencies := parseBooleanArg(args, OPT_TERRAGRUNT_INCLUDE_EXTERNAL_DEPENDENCIES, false)
+	includeExternalDependencies := parseBooleanArg(args, OPT_TERRAGRUNT_INCLUDE_EXTERNAL_DEPENDENCIES, func() bool { _, b := os.LookupEnv("TERRAGRUNT_INCLUDE_EXTERNAL_DEPENDENCIES"); return b }())
```
Comment (brikis98) [inline cli/args.go:106]: Why use a self-calling function here?

File: configstack/stack.go [hunk 2]
```diff
@@ -81,12 +81,6 @@ func (stack *Stack) Run(terragruntOptions *options.TerragruntOptions) error {
 func (stack *Stack) summarizePlanAllErrors(terragruntOptions *options.TerragruntOptions, errorStreams []bytes.Buffer) {
 	for i, errorStream := range errorStreams {
 		output := errorStream.String()
-
```
Comment (brikis98) [inline configstack/stack.go:84]: This shouldn't be part of this PR?

File: docs/_docs/04_reference/cli-options.md [hunk 3]
```diff
@@ -335,28 +335,42 @@ This should allow you to run `import` on the module and work around those Terraf
...
-- [terragrunt-strict-include](#terragrunt-strict-include)
-- [terragrunt-ignore-dependency-order](#terragrunt-ignore-dependency-order)
-- [terragrunt-ignore-external-dependencies](#terragrunt-ignore-external-dependencies)
-- [terragrunt-include-external-dependencies](#terragrunt-include-external-dependencies)
-- [terragrunt-parallelism](#terragrunt-parallelism)
-- [terragrunt-debug](#terragrunt-debug)
-- [terragrunt-check](#terragrunt-check)
-- [terragrunt-hclfmt-file](#terragrunt-hclfmt-file)
-- [terragrunt-override-attr](#terragrunt-override-attr)
+- [CLI commands](#cli-commands)
```
Comment (brikis98) [inline docs/_docs/04_reference/cli-options.md:338]: This shouldn't be part of this PR?

File: docs/_docs/04_reference/cli-options.md [hunk 4]
```diff
@@ -512,7 +526,8 @@ directories.
 
 **CLI Arg**: `--terragrunt-ignore-dependency-order`
 
-When passed in, ignore the depedencies between modules when running `*-all` commands.
+**CLI Arg**: `--terragrunt-include-external-dependencies`</br>
+**Environment Variable**: `TERRAGRUNT_INCLUDE_EXTERNAL_DEPENDENCIES` (set to `false`)</br>
```
Comment (brikis98) [inline docs/_docs/04_reference/cli-options.md:530]: Does this flag apply here?

**Review by brikis98 — COMMENTED** [review]
Thanks for the PR!

**Comment by elebertus** [comment]
@brikis98 sorry about the hassle on the strange commit for `configstack/stack.go` and my poorly configured vscode plugins for markdown.

I think all your comments are addressed now.

**Review by brikis98 — APPROVED** [review]
OK, great, thanks! Code changes LGTM. I'll kick off tests now.

**Comment by brikis98** [comment]
Tests passed! Merging now.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.29.9
